packages/amplify-cli/src/commands/plugin/add.ts

```diff
@@ -1,19 +1,110 @@
 import {
     addUserPluginPackage,
     addExcludedPluginPackage as addFromExcluded,
     confirmAndScan,
 } from '../../plugin-manager';
 import inquirer, { InquirerOption, EXPAND } from '../../domain/inquirer-helper';
 import { AddPluginError } from '../../domain/add-plugin-result';
+import { normalizePluginDirectory } from '../../plugin-helpers/scan-plugin-platform';
 
 const NEWPLUGINPACKAGE = 'A new plugin package';
-
+const CANCAL = 'cancel';
 
 export async function run(context: Context) {
+  if (context.input.subCommands && context.input.subCommands.length > 1) {
+    const input = context.input.subCommands[1];
+    const { excluded } = context.pluginPlatform;
+    if (excluded[input] && excluded[input].length > 0) {
+      const { confirmed } = await inquirer.prompt({
+        type: 'confirm',
+        name: 'confirmed',
+        message: `Add from previously removed ${input} plugin`,
+        default: true,
+      });
+      if (confirmed) {
+        await addExcludedPluginPackage(context, excluded[input]);
+      } else {
+        await resolvePluginPathAndAdd(context, input);
+      }
+    } else {
+      await resolvePluginPathAndAdd(context, input);
+    }
+  } else {
+    await promptAndAdd(context);
+  }
+}
+
+async function resolvePluginPathAndAdd(context: Context, inputPath: string) {
+  const pluginDirPath = await resolvePluginPackagePath(context, inputPath);
+  if (pluginDirPath) {
+    addNewPluginPackage(context, pluginDirPath);
+  }
+}
+
+async function resolvePluginPackagePath(context: Context, inputPath: string):
+Promise<string | undefined> {
+  let result;
+
+  const { pluginPlatform } = context;
+  let searchDirPaths = [
+    constants.ParentDirectory,
+    constants.LocalNodeModules,
+    constants.GlobalNodeModules,
+    process.cwd(),
+  ];
+  searchDirPaths = searchDirPaths.filter(dirPath =>
+    !pluginPlatform.pluginDirectories.includes(dirPath.toString()));
+  searchDirPaths = searchDirPaths.concat(pluginPlatform.pluginDirectories);
+
+  const candicatePluginDirPaths = searchDirPaths.map(dirPath =>
+    path.normalize(path.join(normalizePluginDirectory(dirPath), inputPath)),
+  ).filter(pluginDirPath =>
+    fs.existsSync(pluginDirPath) && fs.statSync(pluginDirPath).isDirectory(),
+  );
+
+  if (candicatePluginDirPaths.length === 0) {
+    context.print.error('Can not locate the plugin package.');
+    result = await promptForPluginPath();
+  } else if (candicatePluginDirPaths.length === 1) {
+    context.print.green('Plugin package found.');
+    context.print.blue(candicatePluginDirPaths[0]);
+    const { confirmed } = await inquirer.prompt({
+      type: 'confirm',
+      name: 'confirmed',
+      message: `Please confirm to add the plugin package to your Amplify CLI.`,
+      default: true,
+    });
+    if (confirmed) {
+      result = candicatePluginDirPaths[0];
+    }
+  } else if (candicatePluginDirPaths.length > 1) {
+    context.print.warning('Multiple plugins with the package name are found.');
```

Comment:
We should output what packages is a duplicate: "Multiple plugins found with the same package name: {packageName}".

Reply:
the package name is the user's input. and below this line is the selection with all the package paths prompted to the user